apps/api-extractor/src/generators/PackageTypingsGenerator.ts

```diff
@@ -0,0 +1,402 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+/* tslint:disable:no-bitwise */
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+
+import { ExtractorContext } from '../ExtractorContext';
+import IndentedWriter from '../IndentedWriter';
+import TypeScriptHelpers from '../TypeScriptHelpers';
+import { Span } from './Span';
+
+/**
+ * An "Entry" is a type definition that we encounter while traversing the
+ * references from the package entry point.  This data structure helps filter,
+ * sort, and rename the entries that end up in the output package typings file.
+ */
+class Entry {
+  /**
+   * The original name of the symbol, as exported from the module (i.e. source file)
+   * containing the original TypeScript definition.
+   */
+  public localName: string;
+
+  /**
+   * The localName, possibly renamed to ensure that all the top-level exports have unique names.
+   */
+  public uniqueName: string | undefined = undefined;
+
+  /**
+   * The compiler symbol where this type was defined, after following any aliases.
+   */
+  public followedSymbol: ts.Symbol;
+
+  /**
+   * If true, this entry should be emitted using the "export" keyword instead of the "declare" keyword.
+   */
+  public exported: boolean = false;
+
+  private _sortKey: string|undefined = undefined;
+
+  public getSortKey(): string {
+    if (!this._sortKey) {
+      if (this.localName.substr(0, 1) === '_') {
+        // Removes the leading underscore, for example:  "_example" --> "example*"
+        // This causes internal definitions to sort alphabetically with regular definitions.
+        // The star is appended to preserve uniqueness, since "*" is not a legal  identifier character.
+        this._sortKey = this.localName.substr(1) + ' ';
```

Comment:
>' ' [](start = 51, length = 3)

Is this the star you talked about in the comment?

Reply:
Good point

---
In reply to: [152430933](https://github.com/Microsoft/web-build-tools/pull/448#discussion_r152430933) [](ancestors = 152430933)